File: Jardin De infantes/DAL/Servicios/Idioma/dalIdioma.vb
Imports EE
Imports System.Data.SqlClient
Imports MP
Public Class dalIdioma

    Private db As Conexion
    Private _cmd As IDbCommand

    Sub New()
        db = New Conexion

    End Sub
    Public Sub Escribir(cadena As String)
        Try

            Dim cmd As SqlCommand = db.CrearComando

            cmd.CommandType = CommandType.Text
            cmd.CommandText = cadena

            cmd.ExecuteNonQuery()

            db.RealizarCommit()

        Catch ex As Exception
            db.RealizarRollBack()
            MsgBox(ex.Message, MsgBoxStyle.Critical, "Error")

        End Try

    End Sub



    Public Function IdatosCompleto_Listar() As List(Of Idioma)

        Try
            Dim cmd As IDbCommand = db.CrearComando
            cmd.CommandType = CommandType.StoredProcedure
            cmd.CommandText = "p_ListarIdioma"
            cmd.ExecuteNonQuery()
            Dim da As New SqlDataAdapter(cmd)
            Dim dt As New DataTable
            da.Fill(dt)
            db.RealizarCommit()
            Dim mapear As New mppIdioma
            Dim listaIdiomas As New List(Of Idioma)
            listaIdiomas = mapear.listar(dt)
            Return listaIdiomas
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Exclamation, "Error al ver Idiomas")
            Return Nothing
        End Try
    End Function


End Class

File: Jardin De infantes/Seguridad/DALseguridad/Idioma/dalIdioma.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace DALseguridad.Idioma
{
    public class dalIdioma
    {
        DataTable dt = new DataTable();
        SqlTransaction tx;
        public DataTable Leer(string texto)
        {
            try
            {
                using (SqlConnection cn = new SqlConnection(dalConexion.Cadena))
                {
                    cn.Open();
                    tx = cn.BeginTransaction();
                    SqlCommand cmd = new SqlCommand(texto, cn, tx);
                    cmd.CommandType = CommandType.Text;
                    cmd.ExecuteNonQuery();
                    SqlDataAdapter da = new SqlDataAdapter(cmd);
                    da.Fill(dt);
                    tx.Commit();
                    cn.Close();
                    return dt;
                }
            }
            catch (Exception e)
            {

                MessageBox.Show("Error en el Backup", e.Message,
                                MessageBoxButtons.OK, MessageBoxIcon.Error); ;
                tx.Rollback();
                return null;
            }

        }

        public void Escribir(string texto)
        {
            try
            {
                using (SqlConnection cn = new SqlConnection(dalConexion.Cadena))
                {
                    cn.Open();
                    tx = cn.BeginTransaction();
                    SqlCommand cmd = new SqlCommand(texto, cn, tx);
                    cmd.CommandType = CommandType.Text;
                    cmd.ExecuteNonQuery();
                    tx.Commit();

                }
            }
            catch (Exception e)
            {
                tx.Rollback();
                MessageBox.Show("Error en el Backup", e.Message,
                                MessageBoxButtons.OK, MessageBoxIcon.Error); ;

            }
        }
    }
}
